R/bin_utils.R

```diff
@@ -0,0 +1,87 @@
+#' Get Bins of Metadata
+#'
+#' Creates unique age/heights-bins for each record to speed up joining
+#'
+#' @param meta_criteria Metadata dataset
+#'
+#'   A metadata dataset that is the same as that in
+#'   `derive_params_growth_age()`/`derive_params_growth_height()`
+#'
+#' @param param Character vector
+#'
+#'   Accepted values: "AGE"/"HEIGHT_LENGTH"
+#'
+#' @family internal
+#' @keywords internal
+#' @return list of breaks and labels for age/height bins
+get_bins <- function(meta_criteria, param) {
+  if (param == "AGE") {
+    unique_vals <- sort(unique(meta_criteria$AGE))
+  } else if (param == "HEIGHT_LENGTH") {
+    unique_vals <- sort(unique(meta_criteria$HEIGHT_LENGTH))
+  }
+  breaks <- c(unique_vals, Inf)
+  labels <- paste0(head(breaks, -1), ",", tail(breaks, -1))
```

Comment:
I get the following messages while running the checks. Could you please consider them accordingly:
- get_bins: no visible global function definition for 'head'
- get_bins: no visible global function definition for 'tail'
- Undefined global functions or variables: age_bins head ht_bins tail